specifyweb/specify/tree_ranks.py

```diff
@@ -305,6 +305,7 @@ def verify_rank_parent_chain_integrity(rank, rank_operation: RankOperation):
             if rank_id is not None:
                 parent_to_children_dict.setdefault(parent_id, []).append(rank_id)
 
-    for parent_id, children in parent_to_children_dict.items():
-        if len(children) > 1 and parent_id is not None:
-            raise TreeBusinessRuleException(f"Parent rank {parent_id} has more than one child rank")
+    if rank_operation == RankOperation.DELETED:
+        for parent_id, children in parent_to_children_dict.items():
+            if len(children) > 1 and parent_id is not None:
+                raise TreeBusinessRuleException(f"Parent rank {parent_id} has more than one child rank")
```

Comment:
hmm i'm kinda thinking the rest of the checks can be enforced by the db itself, no? the ones at lines 300-307. (so you can just get rid of them)

I know you want to "check count of children". but, this can be done more simply. HINT: get a list of parents. what needs to happen for the a node to have multiple children?

also, the child count can be simplified. don't want to